oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java

```diff
@@ -1041,8 +1041,14 @@ private SelectorExecutionPlan getBestSelectorExecutionPlan(
             QueryIndex index = queryIndexes.get(i);
             double minCost = index.getMinimumCost();
             if (minCost > bestCost) {
-                // Stop looking if the minimum cost is higher than the current best cost
-                break;
+                if (minCost - bestIndex.getMinimumCost() < .00001) {
```

Comment:
Should this constant `.00001` be configurable, or do you think hard coding is okay here?

Reply:
Actually we wanted to use equality but as the variable is a double, we can't. In my opinion the minimum cost should be configurable instead of this.
What do you suggest, should we create a jira to make minimum cost configurable?